src/Microsoft.AspNetCore.Server.IISIntegration/IServerVariablesFeature.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Http.Features
+{
+    /// <summary>
+    /// This feature provides access to request server variables set.
+    /// <para>
+    /// This feature is only available when hosting ASP.NET Core in-process with IIS or IIS Express.
+    /// </para>
+    /// </summary>
+    /// <remarks>
+    /// For a list of common server variables available in IIS, see http://go.microsoft.com/fwlink/?LinkId=52471.
+    /// </remarks>
+    public interface IServerVariablesFeature
```

Comment:
One down sides to supporting this very minimal interface, you can't enumerate the server variables.

Reply:
@pan-wang IIS doesn't have an API for enumerating server variables, correct? I was told System.Web makes it work by keeping a static list of variables to try.

Reply:
Yea you can get the list from IIS. Helios does the same thing. It’s not a deal breaker and can be added in the future if required